Lists smileys even when smiley_files table is empty

Sesquipedalian opened the pull request:
Fixes #6918

With this, smileys will still be listed in the various "Smileys and Message Icons" lists even if the smiley_files table is empty.

Note: this PR does not try to correct bad smiley settings during upgrade, but it does allow the admin to fix missing smiley_file data in the UI afterward.

Sesquipedalian committed:
Lists smileys even when smiley_files is empty
Fixes #6918

Signed-off-by: Jon Stovell <jonstovell@gmail.com>

sbulen reviewed (changes requested):
See note regarding dropdown

Sesquipedalian commented:
> See note regarding dropdown

Er, what note? Where?

sbulen commented:
If you see this note on some random unrelated issue, please ignore it over there....   🙄

The browse button doesn't seem to work???
![image](https://user-images.githubusercontent.com/23568484/128650796-61954b4a-e902-4cfb-ae26-9efe527001f4.png)

Sesquipedalian committed:
Removes required attribute from smiley filename select menu
This is necessary so that the admin can submit the form when uploading a new smiley image.

Signed-off-by: Jon Stovell <jonstovell@gmail.com>

Sesquipedalian commented:
Give it a try now, @sbulen. :)

sbulen commented:
Works now, thanks.

Sesquipedalian commented:
Excellent
